client/src/Contexts/AuthContext.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useEffect, useState } from 'react'
+
+import firebase from 'firebase/app'
+import { auth } from '../firebase'
+
+export interface IAuthContext {
+  currentUser: firebase.User | null
+  login: (
+    email: string,
+    password: string
+  ) => Promise<firebase.auth.UserCredential>
+  logout: () => Promise<void>
+  signup: (
+    email: string,
+    password: string
+  ) => Promise<firebase.auth.UserCredential>
+}
+
+export const AuthContext: React.Context<IAuthContext> = React.createContext(
+  {} as IAuthContext
+)
+
+export const AuthProvider: React.FC<React.ReactNode> = ({ children }) => {
```

Comment:
`React.FC` contains children
```suggestion
export const AuthProvider: React.FC = ({ children }) => {
```